api/receiver/receive_results.go

```diff
@@ -169,3 +194,38 @@ func fetchFile(a AppEngineAPI, url string) (io.ReadCloser, error) {
 	}
 	return nil, fmt.Errorf("failed to fetch %s", url)
 }
+
+func handleAzureArtifact(a AppEngineAPI, artifactName string, url string) (int, func(int) (io.ReadCloser, error), error) {
+	log := shared.GetLogger(a.Context())
+	log.Debugf("Detected azure artifact %s", artifactName)
+	artifactZip, err := fetchFile(a, url)
+	if err != nil {
+		log.Errorf("Failed to fetch %s: %s", url, err.Error())
+		return 0, nil, errors.New("Failed to fetch azure artifact")
+	}
+	defer artifactZip.Close()
+	artifact, err := newAzureArtifact(artifactName, artifactZip)
+	if err != nil {
+		log.Errorf("Failed to read zip: %s", err.Error())
+		return 0, nil, errors.New("Invalid artifact contents")
+	}
+	artifactFiles, err := artifact.getReportFiles()
+	if err != nil {
+		log.Errorf("Failed to extract files: %s", err.Error())
+		return 0, nil, errors.New("Invalid artifact contents")
+	}
+	results := len(artifactFiles)
+	log.Debugf("Found %v report files in artifact", results)
+	getFile := func(i int) (io.ReadCloser, error) {
+		zipR, err := artifactFiles[i].Open()
+		if err != nil {
+			return nil, err
+		}
+		buf := new(bytes.Buffer)
+		w := gzip.NewWriter(buf)
+		io.Copy(w, zipR)
+		w.Close()
```

Comment:
Check the returned errors of the two statements above.

Reply:
Done.